src/main/java/htsjdk/samtools/cram/structure/ContainerHeaderIO.java

```diff
@@ -17,29 +17,32 @@
  */
 package htsjdk.samtools.cram.structure;
 
+import htsjdk.samtools.cram.build.CramIO;
 import htsjdk.samtools.cram.io.CRC32OutputStream;
 import htsjdk.samtools.cram.io.CramIntArray;
 import htsjdk.samtools.cram.io.CramInt;
 import htsjdk.samtools.cram.io.ITF8;
 import htsjdk.samtools.cram.io.LTF8;
+import htsjdk.samtools.cram.ref.ReferenceContext;
 import htsjdk.samtools.util.RuntimeIOException;
 
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
 
 class ContainerHeaderIO {
 
-    public boolean readContainerHeader(final Container container, final InputStream inputStream) {
-        return readContainerHeader(2, container, inputStream);
+    public Container readContainerHeader(final InputStream inputStream) {
+        return readContainerHeader(2, inputStream);
     }
 
-    public boolean readContainerHeader(final int major, final Container container, final InputStream inputStream) {
+    public Container readContainerHeader(final int major, final InputStream inputStream) {
         final byte[] peek = new byte[4];
         try {
             int character = inputStream.read();
             if (character == -1)
-                return false;
+                return readContainerHeader(new ByteArrayInputStream((major >= 3 ? CramIO.ZERO_F_EOF_MARKER : CramIO.ZERO_B_EOF_MARKER)));
```

Comment:
This was what the caller did when it received a `false` from this method.

Reply:
Mildly not crazy about the recursion, but this change would make it directly recursive instead of indirectly, allow the otherwise unused method above to be deleted:

```suggestion
                return readContainerHeader(major, new ByteArrayInputStream((major >= 3 ? CramIO.ZERO_F_EOF_MARKER : CramIO.ZERO_B_EOF_MARKER)));
```